rollup.config.js

```diff
@@ -89,6 +89,53 @@ function makeBabelExternal(id) {
   return /^@babel/.test(id);
 }
 
+function makeNodeExternal() {
+  const nodePackageJSONString = readFileSync(
+    "packages/skygear-node-client/package.json",
+    { encoding: "utf8" }
+  );
+  const nodePackageJSON = JSON.parse(nodePackageJSONString);
+
+  const peerDeps = Object.keys(nodePackageJSON["peerDependencies"] || {});
+  if (peerDeps.length > 0) {
+    throw new Error(
+      "@skygear/node-client should not have any peerDependencies"
+    );
+  }
+
+  const deps = Object.keys(nodePackageJSON["dependencies"] || {});
+  const builtins = getBuiltins();
```

Comment:
I suppose this returns a list of built-in modules of NodeJS. I don't think any of them would ever appears in dependencies on package.json, therefore the check should be redundant?

Reply:
Yes, you are right. They wont appear in `dependencies` but it is perfectly fine to use built-in modules of Node in `@skygear/node-client`. The reason why we have `builtins` here is to write the `external` function. The `external` of node has two jobs

1. If the module (`id`) appear in `dependencies`, tell rollup that it is external.
2. Otherwise if the module (`id`) is a node builtin, tell rollup that is is external.